### PR TITLE
[apiap] Backend API edit page

### PR DESCRIPTION
**Pull request by guicassolato**
**What this PR does / why we need it**:

Implements the edit page for backend APIs

**Which issue(s) this PR fixes** 

Closes [THREESCALE-3276](https://issues.jboss.org/browse/THREESCALE-3276)

### CODE REVIEW

File: app/views/provider/admin/backend_apis/edit.html.slim [hunk 1]
```diff
@@ -0,0 +1,8 @@
+h2 Backend API
```
Comment (josemigallas) [inline app/views/provider/admin/backend_apis/edit.html.slim]: This is different in the services' form:
https://github.com/3scale/porta/blob/f01d12881545e6ebba344fb7514191affb949924/app/views/api/services/edit.html.slim#L1
maybe you could change that one too

**Comment by josemigallas** [comment]
I've noticed the ContextSelector doesn't work anymore. The reason is it takes `@service` as argument but that variable doesn't exist in backend_apis. I think this could be fixed by simply passing `@service || @backend_api` to the component.

**Comment by guicassolato** [comment]
> I've noticed the ContextSelector doesn't work anymore.
The context selector is not part of this PR. Anyway, I tested this branch in combination with https://github.com/3scale/porta/pull/1058 and https://github.com/3scale/porta/pull/1077 in Openshift and the context selector works just fine. What am I missing?

File: app/controllers/provider/admin/backend_apis_controller.rb [hunk 2]
```diff
@@ -25,6 +25,10 @@ def create
     end
   end
 
+  def show; end
```
Comment (hallelujah) [inline app/controllers/provider/admin/backend_apis_controller.rb]: Do we need to create them?

**Pull request merged**